EndpointHandler.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+require "settings.php";
+
+// Decodes the data array and returns the value of "claim"
+function getClaim($claim, $data) {
+	
+	$data_array = json_decode($data, true);
+	return $data_array[$claim];
+}
+
+// A class to handle both fetching and sending data to the various endpoints
+class EndpointHandler {
+	
+	private $metadata = "";
+	
+	public function __construct($policy_name) {
+		$this->getMetadata($policy_name);
+	}
+	
+	// Fetches the data at an endpoint using a HTTP GET request
+	public function getEndpointData($uri) {
+
+		$ch = curl_init($uri);
+		curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1); 
+		curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);
+		$resp = curl_exec($ch);
+		curl_close($ch);
+		return $resp;
+	}
+	
+	// Using a HTTP POST request, sends data to the endpoint and receives the response
+	public function postEndpointData($uri, $fields) {
+		
+		//url-ify the data for the POST
+		$fields_string = "";
+		foreach ($fields as $key=>$value) { $fields_string .= $key.'='.$value.'&'; }
+		$fields_string = rtrim($fields_string, '&');
+					
+		$ch = curl_init();
+		curl_setopt($ch,CURLOPT_URL, $uri);
+		curl_setopt($ch,CURLOPT_POST, sizeof($fields));
+		curl_setopt($ch,CURLOPT_POSTFIELDS, $fields_string);
+		curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1); 
+		curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);
+
+		$resp = curl_exec($ch);
+		curl_close($ch);
+		return $resp;
+	}
+	
+	// Given a B2C policy name, constructs the metadata endpoint 
+	// and fetches the metadata from that endpoint
+	public function getMetadata($policy_name) {
+		require "settings.php";
```

Comment:
Fixed by adding in autoloader.

---
In reply to: [71211569](https://github.com/AzureAD/active-directory-b2c-wordpress-plugin-openidconnect/pull/3#discussion_r71211569) [](ancestors = 71211569)